codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -460,7 +469,7 @@ void cancel() {
                 if (frame == null) {
                     break;
                 }
-                writeError(frame, streamError(stream.id(), INTERNAL_ERROR,
+                writeError(frame, streamError(stream.id(), INTERNAL_ERROR, cause,
                                               "Stream closed before write could take place"));
             }
```

Comment:
I'm not sure this really matters the way things are setup but should we set `cancelled = false` after the `cancel` request has been processed?